srml/contract/src/lib.rs

```diff
@@ -262,13 +288,20 @@ decl_event! {
 	pub enum Event<T>
 	where
 		<T as balances::Trait>::Balance,
-		<T as system::Trait>::AccountId
+		<T as system::Trait>::AccountId,
+		<T as system::Trait>::Hash
 	{
 		/// Transfer happened `from` -> `to` with given `value` as part of a `message-call` or `create`.
 		Transfer(AccountId, AccountId, Balance),
 
 		/// Contract deployed by address at the specified address.
-		Created(AccountId, AccountId),
+		Instantiated(AccountId, AccountId),
```

Comment:
> However, it's not that simple. Note that this is wrapped in decl_event! macro, which does not support record style enum variants.

Maybe it should support this? Sounds like a great, precise ticket to me.

Is this external API or could we fix this later? In that case, let's put up a ticket for the `decl_event!` and the API changed described and FIXME-Link it here?

Reply:
I'm not entirely sure, but it sounds like this is a module-private API, however, it might expose some serialization which might be desirable to keep the same. Although this shouldn't be a problem since tuple and record types are equivalent in that sense, the latter one are just have identifiers (observable only in Rust source code).

I think, FIXME link here is not required since basically SRML contains a lot of these cases and I'd prefer just have a general ticket which takes care of each case!